File: quickstart.py
# quickstart.py
import time
from sync import Sync
from sync.common import Audio, GenerationOptions, Video
from sync.core.api_error import ApiError

# ---------- UPDATE API KEY ----------
# Replace with your Sync.so API key
api_key = "sk-JAP3fotwQ8KAKZ4iKYf_yg.VGFw-aMESDWTPMz0k_0Dh39Iui3wUIJU" 

def generate_vid(audio_url):

    video_url = "https://drive.google.com/file/d/1OVjzmRHNLRKp-q0X6I5kWlcCI2LcYHoj/view?usp=sharing"
    audio_url = "https://drive.google.com/file/d/1-5E2Zo2YYXzm3eJgo7o2d83kJMDgQ59A/view?usp=sharing"

    client = Sync(
        base_url="https://api.sync.so", 
        api_key=api_key
    ).generations

    print("Starting lip sync generation job...")

    try:
        response = client.create(
            input=[Video(url=video_url),Audio(url=audio_url)],
            model="lipsync-2",
            options=GenerationOptions(sync_mode="cut_off"),
        )
    except ApiError as e:
        print(f'create generation request failed with status code {e.status_code} and error {e.body}')
        exit()

    job_id = response.id
    print(f"Generation submitted successfully, job id: {job_id}")

    generation = client.get(job_id)
    status = generation.status
    while status not in ['COMPLETED', 'FAILED']:
        print('polling status for generation', job_id)
        time.sleep(10)
        generation = client.get(job_id)
        status = generation.status

    if status == 'COMPLETED':
        print('generation', job_id, 'completed successfully, output url:', generation.output_url)
    else:
        print('generation', job_id, 'failed')
    return generation.output_url

generate_vid("")

File: quickstart.ts

// quickstart.ts
import { SyncClient, SyncError } from "@sync.so/sdk";

// ---------- UPDATE API KEY ----------
// Replace with your Sync.so API key
const apiKey = "sk-JAP3fotwQ8KAKZ4iKYf_yg.VGFw-aMESDWTPMz0k_0Dh39Iui3wUIJU";

// ----------[OPTIONAL] UPDATE INPUT VIDEO AND AUDIO URL ----------
// URL to your source video
const videoUrl = "https://drive.google.com/file/d/11yawaobLN53grrGnLi6h14M1JpXLxvkx/view?usp=sharing"
// URL to your audio file
const audioUrl =  "https://drive.google.com/file/d/1TEFhcdEYFpA57xjew025sdqB175xoLu-/view?usp=sharing"
// ----------------------------------------

const client = new SyncClient({ apiKey });

async function main() {
    console.log("Starting lip sync generation job...");

    let jobId: string;
    try {
        const response = await client.generations.create({
            input: [
                {
                    type: "video",
                    url: videoUrl,
                },
                {
                    type: "audio",
                    url: audioUrl,
                },
            ],
            model: "lipsync-2",
            options: {
                sync_mode: "cut_off",
            },
            outputFileName: "quickstart"
        });
        jobId = response.id;
        console.log(`Generation submitted successfully, job id: ${jobId}`);
    } catch (err) {
        if (err instanceof SyncError) {
            console.error(`create generation request failed with status code ${err.statusCode} and error ${JSON.stringify(err.body)}`);
        } else {
            console.error('An unexpected error occurred:', err);
        }
        return;
    }

    let generation;
    let status;
    while (status !== 'COMPLETED' && status !== 'FAILED') {
        console.log(`polling status for generation ${jobId}...`);
        try {
            await new Promise(resolve => setTimeout(resolve, 10000));
            generation = await client.generations.get(jobId);
            status = generation.status;
        } catch (err) {
            if (err instanceof SyncError) {
                console.error(`polling failed with status code ${err.statusCode} and error ${JSON.stringify(err.body)}`);
            } else {
                console.error('An unexpected error occurred during polling:', err);
            }
            status = 'FAILED';
        }
    }

    if (status === 'COMPLETED') {
        console.log(`generation ${jobId} completed successfully, output url: ${generation?.outputUrl}`);
    } else {
        console.log(`generation ${jobId} failed`);
    }
}

main();